src/aznfswatchdog

```diff
@@ -307,13 +307,49 @@ run_daily_once()
     fi
 }
 
+#
+# We only use lowercase single word names for distro id:
+# ubuntu, centos, redhat, sles.
+#
+canonicalize_distro_id()
+{
+    local distro_lower=$(echo "$1" | tr '[:upper:]' '[:lower:]')
+
+    # Use sles for SUSE/SLES.
+    if [ "$distro_lower" == "suse" ]; then
+        distro_lower="sles"
+    fi
+
+    echo "$distro_lower"
+}
+
+detect_aznfs_version()
+{
+    local distro="$distro_id"
+
+    sleep 45
```

Comment:
why?

Reply:
example for sles, it takes few seconds after which we are able to fetch the right version number, otherwise it is blank output. For similar reason only we deferred auto-update to after 60 seconds to starting watchdog. So, I made this a function a background process.

Reply:
let's discuss